contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/udfs/HttpHelperFunctions.java

```diff
@@ -189,6 +191,8 @@ public void eval() {
         rowWriter.start();
         if (jsonLoader.parser().next()) {
           rowWriter.save();
+        } else {
```

Comment:
Should we explicitly close the `results` `InputStream` here as well?  Would mind testing this on a query that produces multiple batches?

Reply:
@cgivre

1. The JsonLoader closes the input streams it's been working off of when it is closed so I don't think so.
2. Multiple batch datasets do not work with these UDFs yet from what I recall? I think @vdiravka continues to work on that, perhaps he can comment on the closing of the JsonLoader here.

Reply:
From my recollection, this function does handle the multiple batches.  It was the `convert_fromJSON` that @vdiravka was working on.

Reply:
@cgivre yes, you're right. I tried a couple of things. First I provided a JSON response that would normally produce 64k+1 rows if queried to http_get but it looked to me like it was being handled in a single batch since, I guess, the row count of the query is still 1. I then wrote a query to `SELECT http_get(some simple JSON)` from a mock table containing 64k+1 rows. This overwhelms the okhttp3 mock server and fails with a timeout. I'm not sure if there some other test to try here?

Reply:
I fed the http_get function a string containing 50 million little JSON objects from sequence {"foo": 1} {"foo": 2} {"foo": 3}... and it got through it (took about 45s). I just don't know if that answers the right question.

Reply:
That works for me :-)